torch/_dynamo/skipfiles.py

```diff
@@ -29,31 +29,106 @@
 import typing
 import unittest
 import weakref
+from typing import Optional
 
 import torch
 import torch._inductor.test_operators
 import torch.distributed
 import torch.utils._content_store
 
-from . import comptime, config, external_utils
+from . import comptime, external_utils
 
 """
 A note on skipfiles:
 
-Dynamo consults this file to determine whether code should be compiled or skipped.
+Dynamo consults this file to determine whether function should be inlined or skipped.
 
 A skip applies at the frame boundary, meaning dynamo either triggers a graph break
-at the beginning of the frame or attempts to trace the whole frame.  When skipping
+at the beginning of the frame or attempts to trace/inline the whole frame. When skipping
 a frame, recursively called frames are still traced by dynamo unless also skipped.
 
 Skipfiles (skipped at the file level instead of function level) still apply on a
 frame-by-frame boundary as dynamo traces, but apply to all functions in that file.
 
 @skip is a helper decorator that can be applied to your function to cause it to be
 included here.
+
+Dynamo skip/inline rules are defined as follows:
+* Inline is the default behavior and will be used unless explicitly skipped.
+* Dynamo has two SKIPLIST: BUILTIN_SKIPLIST and THIRDPARTY_SKIPLIST.
+    * BUILTIN_SKIPLIST contains builtin python modules, such as abc, collections, etc.
+    * THIRDPARTY_SKIPLIST contains common third party libraries, such as numpy, pandas, etc.
+* Functions in these two SKIPLISTs are always skipped, except when they are explicitly
+    put into the two ALLOWLIST: FILENAME_ALLOWLIST and SUBMODULE_ALLOWLIST.
+* PyTorch(torch) is in the BUILTIN_SKIPLIST by default, but there are many cases
+    where we want inline the functions under torch namespace. We should add them
+    into FILENAME_ALLOWLIST or SUBMODULE_ALLOWLIST to make dynamo inline those functions.
```

Comment:
I'm not sure, just reading this, if ALLOWLIST refers to allowed in graph or not, which would imply you don't inline.

Reply:
Haha, actually this is a terminology confusion or misuse. Everything ALLOW in ```skipfiles.py``` means not skip(inline), this is orthogonal with ALLOW in ```allow_functions.py``` which decided if wrapping it as ```TorchVariable```. I was thinking to name the above lists as  ```FORCE_INLINE_FILENAME_LIST``` and ```FORCE_INLINE_SUBMODULE_LIST``` to avoid confusion. Let's rename it.